specs/eventing/data-plane.md

```diff
@@ -1,125 +1,167 @@
-# Knative Eventing Data Plane Contracts
+# Knative Eventing Data Plane Contract
 
 ## Introduction
 
-Developers using Knative Eventing need to know what is supported for delivery to
-user provided components that receive events. Knative Eventing defines contract
-for data plane components and we have listed them here.
-
-## Conformance
+Late-binding event senders and receivers (composing applications using
+configuration) only works when all event senders and recipients speak a common
+protocol. In order to enable wide support for senders and receivers, Knative
+Eventing extends the [CloudEvents HTTP
+bindings](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md)
+with additional semantics for the following reasons:
+
+- Knative Eventing aims to enable highly-reliable event processing workflows. As
+  such, it prefers duplicate delivery to discarded events. The CloudEvents spec
+  does not take a stance here.
```

Comment:
I found this a bit too restrictive. In particular, the IMC-based broker offers no such guarantees.

Reply:
It does prefer them :) They are just not persisted anywhere, but point taken :)

Reply:
There are use cases for at-most-once delivery guarantee, which prefer losing events to duplicate delivery.